### ISSUE TITLE
windows e2e test failing

### ISSUE THREAD

**Issue by mweibel** [kubernetes-sigs/cluster-api-provider-azure#3276]
**Which jobs are flaky:**
`pull-cluster-api-provider-azure-e2e`

**Which tests are flaky:**
` Creating a VMSS cluster [REQUIRED] with a single control plane node and an AzureMachinePool with 2 Linux and 2 Windows worker nodes`

**Testgrid link:**
https://prow.k8s.io/job-history/gs/kubernetes-jenkins/pr-logs/directory/pull-cluster-api-provider-azure-e2e

**Reason for failure (if possible):**
I didn't look into it all but it looks like the CNI is not set up correctly:
- https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-azure/3243/pull-cluster-api-provider-azure-e2e/1634532750398590976/artifacts/clusters/capz-e2e-e8d5c8-vmss/default/web-windowsp441oz-594ff79bf4-gz6ww/pod-describe.txt
- https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-azure/3241/pull-cluster-api-provider-azure-e2e/1634499211770728448/artifacts/clusters/capz-e2e-jlgv95-vmss/default/web-windowspobit0-8d57f97cd-sr8m6/pod-describe.txt
- https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-azure/3243/pull-cluster-api-provider-azure-e2e/1634499120557199360/artifacts/clusters/capz-e2e-hwt6fs-vmss/default/web-windowsanl76s-85567db77d-5jxfx/pod-describe.txt
- https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-azure/3272/pull-cluster-api-provider-azure-e2e/1634396259517206528/artifacts/clusters/capz-e2e-5wmgf5-vmss/default/web-windowse133hz-5ffcd94f8b-gn47v/pod-describe.txt

```
Failed to create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox "d65c8cf27f3ca3026df4c23391a607f060fc015dd614c9f0f15fd25a2584170f": plugin type="calico" failed (add): global strict affinity should not be false for Windows node
```

I'm not sure why it works sometimes and sometimes not. But it's CNI so it might well depend on timing and that's why the test is flaky. The successful ones don't seem to have the web-windows DS at all and I haven't yet investigated why and what the difference is. 

This is after #3251 got fixed. 

/kind flake

**Comment by CecileRobertMichon** [comment]
This one is a known flake, duplicate of #3227

**Comment by mweibel** [comment]
Oh, didn’t see/remember that obe. Thanks and sorry for the noise.
